packages/@react-spectrum/steplist/src/StepList.tsx

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2020 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {classNames, useDOMRef, useStyleProps} from '@react-spectrum/utils';
+import {DOMRef} from '@react-types/shared';
+import {filterDOMProps} from '@react-aria/utils';
+import {getChildrenSafe} from './utils';
+import React, {Ref} from 'react';
+import {SpectrumStepListProps} from '@react-types/steplist';
+import {StepListItem} from './StepListItem';
+import styles from '@adobe/spectrum-css-temp/components/steplist/vars.css';
+import {useProviderProps} from '@react-spectrum/provider';
+import {useStepList} from '@react-aria/steplist';
+import {useStepListState} from '@react-stately/steplist';
+
+function StepList<T extends object>(props: SpectrumStepListProps<T>, ref: DOMRef) {
+  props = useProviderProps(props);
+  let {styleProps} = useStyleProps(props);
+  let state = useStepListState(props);
+  let ariaProps = useStepList(); // useStepList(props, state);
```

Comment:
Yep, this should take props and state. Probably a good one to look at is ActionGroup. It has some similar patterns to what I think you need.

Reply:
Thanks!  I will move the use of `filterDomProps` into `useStepList` similar to how ActionGroup does.  I'm still not sure what `useStepList` needs to do with the `state` in this case though, as it doesn't seem like there's much that needs to be done in `useStepList`, most of the aria values are set on a per item basis in `useStepListItem`.  Is there something else I'm missing?

Reply:
Our pattern in the past has been to just pass along state to the children and render appropriately https://github.com/adobe/react-spectrum/blob/main/packages/%40react-spectrum/actiongroup/src/ActionGroup.tsx#L102
While StepList only has one selection mode, "single", we should probably make use of useSelectableItem like this as well https://github.com/adobe/react-spectrum/blob/1d9905e7c8308cfbb2a1e03bfc22b2dce89cbb12/packages/%40react-aria/actiongroup/src/useActionGroupItem.ts#L43

I think there is a chance that individual steps in a steplist can be disabled btw

Reply:
In the recent commits, I changed `useStepList` to take props, and moved the `filterDomProps` into `useStepList` but I'm still don't know of  anything that needs to be done in `useStepList` that requires the state.